src/Akka.Streams.Kafka/Messages/CommittableOffsetBatch.cs

```diff
@@ -160,7 +167,7 @@ internal CommittableOffsetBatch Filter(Predicate<GroupTopicPartition> p)
         var newOffsets = OffsetsAndMetadata.Where(o => p(o.Key))
             .ToImmutableDictionary(o => o.Key, o => o.Value);
         var newCommiters =
-            Offsets.ToImmutableDictionary(c => c.GroupTopicPartition, v => CommitterFor(v.GroupTopicPartition));
+            Offsets.ToImmutableDictionary(c => c.Key, c => CommitterFor(c.Key));
```

Comment:
One of the reasons why we needed a dictionary here was to ensure that only a single committer per `GroupTopicPartition` could be found - if we had multiple `GroupTopicPartitionOffset`s in the hashset this would have blown up.